### ISSUE TITLE
Error: Undefined offset: 9

### ISSUE THREAD

**Issue by michaelfromtheoutfit** [markhuot/craftql#210]
Hey Mark!

**Craft Version: 3.1.1**
**CraftQl Version: 1.2.1**

Running into an issue that I believe surfaced after I added tags to my build. Suddenly all of my queries are returning:

`
{
  "error": "Undefined offset: 9"
}
`

I went over to the `api/debug` endpoint you mentioned in another post and this is what I am getting:

```
1. in [redacted]/vendor/markhuot/craftql/src/Repositories/TagGroup.php at line 21

function load() {
      foreach (Craft::$app->tags->allTagGroups as $group) {
            if (!isset($this->groups[$group->id])) {
              $this->groups[$group->id] = $group;
            }
        }
    }
 
    function get($id) {
      return $this->groups[$id];
    }
 
    function all() {
        return $this->groups;
    }
 
}
```
            
Thanks in advance for any assistance and thanks for an another awesome plugin.

MW

**Comment by markhuot** [comment]
Ah, yea, this is because you have a field pointing to a tag group that doesn't exist any more. It's a weird thing about Craft. If you have a tag field pointing to group 9 and then you delete group 9 the field isn't updated. It's still pointing to 9. I'll get a workaround in there for this.

**Comment by michaelfromtheoutfit** [comment]
I thought it might be something like this (because there was a similar request that had to do with categories) but I've only created one tag group and haven't removed any.

If I look at the db directly in craft_taggroups, there is just the one with the autoincrement id field set to 1. So I'm not sure exactly where the 9 is coming from.

Weird, right?

**Comment by markhuot** [comment]
Is there any way you can share a DB with me so I can take a peek? If not I'll see if I can recreate this on my own.

**Comment by michaelfromtheoutfit** [comment]
Ah, I don't think I can share the full DB but I could share specific tables if that would be helpful. All of the content/entry/tags tables are just placeholder content at this point. If there's a list of tables you'd like to look at, just shoot me a message with that and where you'd like me to send the mysqldump.

**Comment by markhuot** [comment]
Sorry, I figured out the issue here. It's a result of a Craft 3.1 change to use `uid` instead of `id`. I have a fix on `dev-master` and after some testing will get it released to a stable version.

**Comment by michaelfromtheoutfit** [comment]
Thanks so much @markhuot! Really appreciate the help.

**Comment by markhuot** [comment]
This is fixed in 1.3.1!
